trek.js

```diff
@@ -0,0 +1,155 @@
+const URL = 'https://ada-backtrek-api.herokuapp.com/trips/';
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trips-list');
+  tripList.empty();
+
+  axios.get(URL)
+  .then((response) => {
+    console.log(response);
+    reportStatus(`Successfully loaded ${response.data.length} trips`);
+    response.data.forEach((trip) => {
+      tripList.append(`<li id="${trip.id}">${trip.name}</li>`);
+    });
+  })
+  .catch((error) => {
+    reportStatus(`Encountered an error while loading trips: ${error.message}`);
+    console.log(error);
+  });
+};
+
+const individualTrip = (id) => {
+  console.log(id);
+  reportStatus("Loading trip details...");
+  const reserve = $('#reservation-form');
+  const trip = $('#trip');
+  trip.empty();
+  reserve.empty();
+
+  let urlTrip = URL + id;
+  axios.get(urlTrip)
+  .then((response) => {
+
+    reportStatus(`Successfully loaded trip to: ${response.data.name}`);
+
+    trip.append(
```

Comment:
It might make sense to break the code to render trip details out into a separate function - as is, this function is doing two things (making the request, and rendering data). See also http://callbackhell.com/